src/targets/atmega32u4-generic/main.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: MIT
+ * SPDX-FileCopyrightText: 2022 Rafael Silva <perigoso@riseup.net>
+ */
+
+#include <avr/interrupt.h>
+#include <avr/io.h>
+#include <avr/power.h>
+#include <avr/wdt.h>
+
+#include <string.h>
+
+#include "util/data.h"
+#include "util/types.h"
+
+#include "platform/avr/boot.h"
+#include "platform/avr/hal/hid.h"
+#include "platform/avr/usb.h"
+
+void main()
+{
+	/* Disable watch dog */
+	MCUSR &= ~(1 << WDRF);
+	wdt_disable();
+
+	/* Disable clock division */
+	clock_prescale_set(clock_div_1);
+
+	struct hid_hal_t hid_hal;
+	u8 info_functions[] = {
+		OI_FUNCTION_VERSION,
+		OI_FUNCTION_FW_INFO,
+		OI_FUNCTION_SUPPORTED_FUNCTION_PAGES,
+		OI_FUNCTION_SUPPORTED_FUNCTIONS,
+	};
+
+	/* create protocol config */
+	struct protocol_config_t protocol_config;
```

Comment:
I don't know what C version do openinput use, but this is possible on newer versions for zero initialization:
```suggestion
	struct protocol_config_t protocol_config = {};
```

Reply:
I don't think we're freezing to a particular standard right now, but we do or will, it would be one that supports this

Either way, static uninitialized variables end up on the .bss and are set to 0 by default, at least on our arm targets